### ISSUE TITLE
Yearly progression: past year is empty

### ISSUE THREAD

**Issue by j0nes2k** [thomaschampagne/stravistix#112]
The yearly progression on the Athlete's page is a great idea. Although I have logged rides in the last year, the data from past year is completely empty. "Force refresh" does not change anything. 

Are there any prerequisites so that this feature works?

<img width="413" alt="bildschirmfoto 2015-08-10 um 18 14 23" src="https://cloud.githubusercontent.com/assets/1328479/9176468/d3405c38-3f8b-11e5-81d2-6ec1a18f8ea2.png">

**Comment by thomaschampagne** [comment]
Hey jones,

Do you have javascript errors in browser? F12 on chrome to get developper
tools,

Thx !

Tom

**Comment by j0nes2k** [comment]
Hey Tom,

sorry, no error messages showing up. Do you want me to check anything else? Otherwise I will install the develop version and start debugging from there (will probably take a bit until I find the time though).

**Comment by tazmanska** [comment]
Hey jones,

Do you have any cycling activities logged before 08/10 (08/12 for today) in 2014?

**Comment by j0nes2k** [comment]
@tazmanska Yes, there are several cycling activities in 2014.

I have just stepped into this. In AthleteStatsModifier.init() I get an array of 29 activities, however the earliest activity dates back to 2014-11-01. Force refresh does not change anything either.

**Comment by tazmanska** [comment]
OK, so everything looks good. The previous year progression shows activities "to date", so for 08/17/2015 it will compare activities from 01/01/2015-08/17/2015 with the activities from 01/01/2014-08/17/2014. Check if you have any cycling activities logged within this date range. If you do, there is a problem - could you share your athlete id with us?

**Comment by thomaschampagne** [comment]
@j0nes2k This ticket can closed or not?

**Comment by j0nes2k** [comment]
@thomaschampagne Sorry, I did not find the time to have a look at this bug yet. However I have just checked if this was fixed in Stravistix v1, but the error still persists: there are some "undefined" strings and no comparison to past year is available. So this ticket should be kept open.

<img width="420" alt="bildschirmfoto 2015-10-18 um 18 08 10" src="https://cloud.githubusercontent.com/assets/1328479/10564873/5936aa14-75c3-11e5-8764-5c503d92232f.png">

**Comment by tazmanska** [comment]
@j0nes2k I found the cause of 'undefineds'. The problem with no comparison to past year is stil a mystery.

**Comment by rdbahm** [comment]
I am experiencing this same issue - Only a small subset of last year's rides are appearing. The number of miles last year should be closer to 1700 miles. See below for what I currently see.

As experienced by another user, I get no errors in the JS console.

Here is a link to my profile: https://www.strava.com/athletes/3789055

![strava](https://cloud.githubusercontent.com/assets/4553626/11345821/59b58830-91ce-11e5-8609-be3529f871e1.PNG)

**Comment by tazmanska** [comment]
@j0nes2k , @rdbahm  - could you go to http://www.jonathanokeeffe.com/strava/annualSummary.php, generate yours annual summary, save them as html and send it to me via e-mail?

**Comment by rdbahm** [comment]
Included page saved as HTML in a ZIP archive.

**Comment by hauke-w** [comment]
I have the same issue: not all rides from 2014 are included.
I noticed that the start of the graph changes with every new activity. When I looked at annual summary (http://www.jonathanokeeffe.com/strava/annualSummary.php) I saw that the last considered acivity in the graph is the 599th activity (where activity #1 is my most recent activity). So I believe the cause of the issue is that only the last 599 activities are processed.
